### ISSUE TITLE
Skill list bug

### ISSUE THREAD

**Issue by DigiBear** [zombieFox/awesomeSheet#157]
It seems that the Profession, Preform and the Craft skills still show up even if they're empty, just noted with a number after them.

![image](https://user-images.githubusercontent.com/26846158/38461350-bb7953da-3acd-11e8-8864-8634ef570217.png)

seems to be an missing link the a display property

**Comment by DigiBear** [comment]
It seems to me the logic is sound, it shows all skills at this moment except for all untrained skills that can only be used when you're trained in them.

The logic however pushes all craft and perform entries out and appends the corresponding number for the pushed skill slot even when the entries are empty.

Seems that upon review it's not necessarily a bug but the logic currently pushes out all craft and perform skills even if they're not in use (aka empty field). It might look nicer to add logic to the skill list along these lines:

If (firstField === empty & & secondField === empty)  {
  firstField = genericSkill (e.g. craft: +2) 
  Hide firstField.textField (remove the appended numeral) 
  Hide secondField
} else if (firstField === filled || secondField === filled) {
  filledField = firstField (making sure whatever field is filled is first in the logic) 
  firstField.pushAsNormal 
  secondField = genericSkill
  hide secondField.textField
} else if (firstField === filled & & secondField === filled) {
  firstField.pushAsNormal
  secondField.pushAsNormal
}

I hope that made sense.. Mobile typing aint too great to begin with.. 

Basically see if any fields are filled, if yes push them of not show generic skill (e.g. Craft or perform without ranks). Might rven be an idea to just always show the generic skill and only push the craft and perform skill if they're filled like trained only skills.

Sorry for the long text. Let me know if my intentions were unclear! 😁

**Comment by DigiBear** [comment]
Sounds good to me!
 It can probably pretty easily be implemented by checking if the nameField.value.length is greater than 0 or not and changing the display based on that. 

You could also create a "hidden skill" that is only shown in display mode not in edit mode. Just a quick craft entry based upon the Int modifier that is nom-ajustable but always there. Then you can literally just choose to show the other crafts by checking if the nameField is filled.

**Comment by DigiBear** [comment]
Pity

**Comment by zombieFox** [comment]
After some investigation the logics ended up being a bit more complex. I now regret even having 2 Craft, Preform and Profession skills! Oh well, it looks more like this now:
```
var variantSkill = function(key) {
  var variantSkill1 = "skills.default." + key + "_1"; // example lookup, real code references other look up functions
  var variantSkill2 = "skills.default." + key + "_2"; // example lookup, real code references other look up functions
  // if skill is trained only
  if (variantSkill1.trained) {
    // if both skill variant names are not entered
    if (variantSkill1.variant_name == "" && variantSkill2.variant_name == "") {
      // if the variant totals are the same
      if (variantSkill1.current == variantSkill2.current) {
        // add a single generic skill with its total
      } else { // if variant totals are not the same
        // if the skill has ranks
        if (variantSkill1.ranks != "") {
          // add a skill with a number prefixed generic name
        };
        // if the skill has ranks
        if (variantSkill2.ranks != "") {
          // add a skill with a number prefixed generic name
        };
      };
    } else { // if either skill variant names are entered
      // if the skill has ranks
      if (variantSkill1.ranks != "") {
        // add a skill with the user entered name or a number prefixed generic name
      };
      // if the skill has ranks
      if (variantSkill2.ranks != "") {
        // add a skill with the user entered name or a number prefixed generic name
      };
    };
  } else { // if skill is not trained only
    // if both skill variant names are not entered
    if (variantSkill1.variant_name == "" && variantSkill2.variant_name == "") {
      // if both skill variant totals are the same
      if (variantSkill1.current == variantSkill2.current) {
        // add a single generic skill with its total
      } else { // if skill variant totals are not the same
        // add a skill with a number prefixed generic name
        // add a skill with a number prefixed generic name
      };
    } else { // if either skill variant names are entered
      // add a skill with the user entered name or a number prefixed generic name
      // add a skill with the user entered name or a number prefixed generic name
    };
  };
};
variantSkill("craft");
variantSkill("perform");
variantSkill("profession");

```

**Comment by zombieFox** [comment]
Feature is now live.
